backend/server.py

```
@@ -71,14 +138,65 @@ def calc():
+        raise HTTPException(500, detail=f'Authentication failed for source {source}')
+    
+    # Store credentials in database
+    questdb_pool = app.state.questdb_pool
+    user_id = get_user_id(logto_token)
+    is_admin = user_is_admin(logto_token)
+
+    add_credentials(questdb_pool, user_id, source, source_token, is_admin)
+
+
+@app.get("/chat")
+async def chat(logto_token: str, query: str, chat_id: str):
```

Ruff (0.14.14)

[warning] Unused function argument: `chat_id`

(ARG001)